src/libraries/System.Linq.AsyncEnumerable/ref/System.Linq.AsyncEnumerable.cs

```diff
@@ -110,6 +110,9 @@ public static partial class AsyncEnumerable
         public static System.Linq.IOrderedAsyncEnumerable<T> Order<T>(this System.Collections.Generic.IAsyncEnumerable<T> source, System.Collections.Generic.IComparer<T>? comparer = null) { throw null; }
         public static System.Collections.Generic.IAsyncEnumerable<TSource> Prepend<TSource>(this System.Collections.Generic.IAsyncEnumerable<TSource> source, TSource element) { throw null; }
         public static System.Collections.Generic.IAsyncEnumerable<int> Range(int start, int count) { throw null; }
+#if NET7_0_OR_GREATER
+        public static System.Collections.Generic.IAsyncEnumerable<T> Range<T>(T start, int count) where T : System.Numerics.IBinaryInteger<T> { throw null; }
+#endif
```

Comment:
Thanks. This should be in its own System.Linq.AsyncEnumerable.netcore.cs file rather than ifdef'd here.